dafni_cli/model/inputs.py

```diff
@@ -0,0 +1,193 @@
+    INPUT_DESCRIPTION_HEADER,
+    INPUT_DESCRIPTION_LINE_WIDTH,
+    INPUT_MAX_HEADER,
+    INPUT_MIN_HEADER,
+    INPUT_MIN_MAX_COLUMN_WIDTH,
+    INPUT_TITLE_HEADER,
+    INPUT_TYPE_COLUMN_WIDTH,
+    INPUT_TYPE_HEADER,
+    TAB_SPACE,
+)
+from dafni_cli.utils import optional_column_new
+
+
+@dataclass
+class ModelDataslot(ParserBaseObject):
+    """Dataclass representing an input dataslot for a DAFNI
+       model
+
+    Attributes:
+        name (str): Name of the slot
+        path (str): Path of the file expected in the container
+        defaults (List[str]): List of default dataset ids
+        required (str): Whether the slot is required
+        description (Optional[str]): Description of the slot if applicable
+    """
+
+    name: str
+    path: str
+    defaults: List[str]
+    required: bool
+    description: Optional[str] = None
+
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("name", "name", str),
+        ParserParam("path", "path", str),
+        ParserParam("defaults", "default"),
+        ParserParam("required", "required"),
+        ParserParam("description", "description", str),
+    ]
+
+
+@dataclass
+class ModelParameter(ParserBaseObject):
+    """Dataclass representing an input parameter for a DAFNI model
+
+    Attributes:
+        name (str): Name of the parameter
+        type (str): Type of the parameter
+        title(str): Title of the parameter
+        required (bool): Whether the parameter is required
+        description (str): Description of the model parameter
+        default (Any): Default value of the parameter
+    """
+
+    name: str
+    type: str
+    title: str
+    required: bool
+    description: str
+    default: Optional[Any] = None
+
+    # TODO: Left over from refactor, may no longer be needed
+    # (printed in format_parameters in ModelInputs but untouched for 2 years)
+    min: Optional[str] = None
+    max: Optional[str] = None
+
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("name", "name", str),
+        ParserParam("type", "type", str),
+        ParserParam("title", "title"),
+        ParserParam("required", "required"),
+        ParserParam("description", "description"),
+        ParserParam("default", "default"),
+        ParserParam("min", "min"),
+        ParserParam("max", "max"),
+    ]
+
+
+@dataclass
+class ModelInputs(ParserBaseObject):
+    """Dataclass representing inputs for a DAFNI model
+
+    Attributes:
+        parameters (List[ModelParameters]): List of parameters for the model
+        dataslots (List[ModelDataslot]): List of dataslots for the model
+    """
+
+    parameters: List[ModelParameter]
+    dataslots: List[ModelDataslot] = field(default_factory=list)
+
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("parameters", "parameters", ModelParameter),
+        ParserParam("dataslots", "dataslots", ModelDataslot),
+    ]
+
+    # TODO: Couldn't we use tabulate or something for this?
```

Comment:
Yes, if there's a tidier way to do all this printing please do

Reply:
Sure, but I wont do it right now, this PR will be big enough as it is 😄